src/xdocs-examples/resources-noncompilable/com/puppycrawl/tools/checkstyle/checks/naming/patternvariablename/Example2.java

```diff
@@ -0,0 +1,23 @@
+/*xml
+<module name="Checker">
+  <module name="TreeWalker">
+    <module name="PatternVariableName">
+        <property name="format" value="^[a-z](_?[a-zA-Z0-9]+)*$"/>
+    </module>
+  </module>
+</module>
+*/
+
+//non-compiled with javac: Compilable with Java17
+package com.puppycrawl.tools.checkstyle.checks.naming.patternvariablename;
+
+// xdoc section -- start
+class Example2 {
+  Example2(Object o1){
+    if (o1 instanceof String STRING) { // violation
+    }
+    if (o1 instanceof Integer num) {
```

Comment:
Please add to all inputs `num_1` to show that no violation is reported if allowed in format

Reply:
Yep Done....